src/parser/Statement.ts

```diff
@@ -27,6 +28,41 @@ export interface Visitor<T> {
     visitLibrary(statement: Library): BrsInvalid;
 }
 
+let statementTypes = new Set<string>([
+    "Assignment",
+    "Expression",
+    "ExitFor",
+    "ExitWhile",
+    "Print",
+    "If",
+    "Block",
+    "For",
+    "ForEach",
+    "While",
+    "Function",
+    "Return",
+    "DottedSet",
+    "IndexedSet",
+    "Increment",
+    "Library",
+    "Dim",
+]);
+
+/**
+ * Returns a boolean of whether or not the given object is a Statement.
+ * @param obj object to check
+ */
+export function isStatement(obj: Expr.Expression | Stmt.Statement) {
+    // This is to play nice with Typescript. We know that implementations of Statements
+    // and Expressions also extend AstNode, but Typescript doesn't know that.
+    let ast = (obj as unknown) as AstNode;
```

Comment:
🤔 I'm not a huge fan of the `(x as unknown) as y` pattern.

> We know that implementations of Statements and Expressions also extend AstNode, but Typescript doesn't know that.

Let's tell TypeScript that then!  We can modify `interface Expression` to `interface Expression extends AstNode` and `interface Statement` -> `interface Statement extends AstNode`.  At that point, `obj` is guaranteed to be an instance of `AstNode`, so there's no casting necessary here 😃

Reply:
Great idea! Just made that change 👍